dali/pipeline/operators/geometric/bb_flip_test.cc

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/test/dali_test_single_op.h"
+
+namespace dali {
+
+namespace {
+
+const int BB_STRUCT_SIZE = 4;
+
+/**
+ * Roi can be represented in two ways:
+ * 1. Upper-left corner, width, height
+ *    (x1, y1,  w,  h)
+ * 2. Upper-left and Lower-right corners
+ *    (x1, y1, x2, y2)
+ *
+ * Both of them have coordinates in image coordinate system
+ * (i.e. 0.0-1.0)
+ */
+struct Roi {
+  float roi[BB_STRUCT_SIZE];
+};
+
+/**
+ * Functor for calculating Roi hash
+ */
+struct RoiHash {
+  std::size_t operator()(Roi const &roi) const noexcept {
+    std::stringstream ss;
+    ss << std::to_string(roi.roi[0]) << std::to_string(roi.roi[1])
+       << std::to_string(roi.roi[2]) << std::to_string(roi.roi[3]);
+    return std::hash<std::string>{}(ss.str());
+  }
+};
+
+
+bool operator==(const Roi &lh, const Roi &rh) noexcept {
+  return RoiHash{}(lh) == RoiHash{}(rh);  //NOLINT
+}
+
+
+std::unordered_map<Roi, Roi, RoiHash> wh_rois = {
+        {{.2,  .2,  .4, .3}, {.4,  .2,  .4, .3}},
+        {{.0,  .0,  .5, .5}, {.5,  .0,  .5, .5}},
+        {{.3,  .2,  .1, .1}, {.6,  .2,  .1, .1}},
+        {{.0,  .0,  .2, .3}, {.8,  .0,  .2, .3}},
+        {{.0,  .0,  .1, .1}, {.9,  .0,  .1, .1}},
+        {{.5,  .5,  .1, .1}, {.0,  .5,  .1, .1}},
+        {{.0,  .6,  .7, .4}, {.3,  .6,  .7, .4}},
+        {{.6,  .2,  .3, .3}, {.1,  .2,  .3, .3}},
+        {{.4,  .3,  .5, .5}, {.1,  .3,  .5, .5}},
+        {{.25, .25, .5, .5}, {.25, .25, .5, .5}},
+};
+
+std::unordered_map<Roi, Roi, RoiHash> two_pt_rois = {
+        {{.2,  .2,  .6,  .5},  {.4,  .2,  .8,  .5}},
+        {{.0,  .0,  .5,  .5},  {.5,  .0,  1.,  .5}},
+        {{.3,  .2,  .4,  .3},  {.6,  .2,  .7,  .3}},
+        {{.0,  .0,  .2,  .3},  {.8,  .0,  1.,  .3}},
+        {{.0,  .0,  .1,  .1},  {.9,  .0,  1.,  .1}},
+        {{.5,  .5,  .6,  .6},  {.4,  .5,  .5,  .6}},
+        {{.0,  .6,  .7,  .9},  {.3,  .6,  1.,  .9}},
+        {{.6,  .2,  .3,  .3},  {.1,  .2,  .4,  .5}},
+        {{.4,  .3,  .9,  .8},  {.1,  .3,  .6,  .8}},
+        {{.25, .25, .75, .75}, {.25, .25, .75, .75}},
+};
+
+using RoiMap = std::unordered_map<Roi, Roi, RoiHash>;
```

Comment:
Why not just use std::array<std::tuple<Roi, Roi>, NUMtestCases> or std::vector? Defining hash function and map to just copy data to plain vector seems like a big overkill.

Reply:
I went with the map, so that there won't be 4 vectors, but `std::array` seems like a nice solution!